### PR TITLE
add workflow to update embedded AppSec rules

### PR DESCRIPTION
**Pull request by RomainMuller**
The workflow simply runs `_tools/rule-pdate/update.sh` and creates a PR with the result.

### CODE REVIEW

File: .github/workflows/update-rules.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        uses: actions/upload-artifact@v3
+        with:
+          name: repo.patch
+          path: ${{ runner.temp }}/repo.patch
+
+  create-pr:
+    needs: update
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
```
Comment (RomainMuller) [inline .github/workflows/update-rules.yml:33]: This is the reason why this is separated from actually creating the patch :)

File: .github/workflows/update-rules.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+jobs:
+  update:
+    runs-on: ubuntu-latest
+    outputs:
+      mutation_happened: ${{ steps.detect.outputs.mutation_happened }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+      - name: Upgrade AppSec Rules
+        run: ./_tools/rules-updater/update.sh
```
[the remark this answers is not in the document]
Comment (RomainMuller) [reply .github/workflows/update-rules.yml]: Ah you're right... Made it possible to specify "latest" there... :)

File: .github/workflows/update-rules.yml (hunk 1 above)
Comment (Julio-Guerra) [reply .github/workflows/update-rules.yml:33]: Very cool GH feature - I wasn't aware of it

File: .github/workflows/update-rules.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        uses: actions/upload-artifact@v3
+        with:
+          name: repo.patch
+          path: ${{ runner.temp }}
+      - name: Apply Patch
+        run: git apply ${{ runner.temp }}/repo.patch
+      - name: Configure git identity
+        run: |-
+          git config user.name "github-actions[bot] on behalf of ${{ github.actor }}"
+          git config user.email "${{ github.actor }}@users.noreply.github.com"
```
Comment (Julio-Guerra) [inline .github/workflows/update-rules.yml:47]: I though the default settings were already these.

Comment (RomainMuller) [reply .github/workflows/update-rules.yml:47]: Last I did this... there was NO default settings, making `git` fall back to interactively prompting for it (which fails).

**Review by Hellzy — APPROVED** [review]

**Pull request merged**